src/graph.rs

```diff
@@ -12,50 +12,116 @@ pub struct Graph<T> {
     edges: HashMap<Rc<T>, HashSet<Rc<T>>>,
 }
 
-impl<T: Eq + PartialEq + Hash> Graph<T> {
+impl<T: Eq + PartialEq + Hash + Clone> Graph<T> {
     pub fn new() -> Graph<T> {
-        todo!();
+        // Should return an empty graph with default values
+        let edges = HashMap::new();
+        Graph { edges }
     }
 
     pub fn vertices(&self) -> Vec<Rc<T>> {
-        todo!();
+        // Iterate through the edges and return the keys in a Vec
```

Comment:
For the example graph,
A->B, A->C.
Your implementation will fail to recognize vertex C